src/engine/serverbrowser.cpp

```diff
@@ -526,8 +470,32 @@ void writeservercfg()
     f->printf("// servers which are connected to or queried get added here automatically\n\n");
     loopv(servers)
     {
-        serverinfo *s = servers[i];
-        f->printf("addserver %s %d %d %s %s %s %s\n", s->name, s->port, s->priority, escapestring(s->sdesc[0] ? s->sdesc : s->name), escapestring(s->authhandle), escapestring(s->flags), escapestring(s->branch));
+        servers[i]->writecfg( *f );
     }
     delete f;
 }
+
+VAR(IDF_PERSIST, hideincompatibleservers, 0, 0, 1);
+void getservers(int server, int prop, int idx)
+{
+    if(server < 0)
+    {
+        // this codes does not really hide stuff, it _removes_ stuff
+        if (hideincompatibleservers)
+        {
+            //servers.erase( std::remove_if(), servers.end()); refuses to work here, dunno why
+            //so for now let's keep the waste
+            vector<serverinfo *> servers_new;
+            std::copy_if( servers.begin(), servers.end(),
+                std::back_inserter( servers_new ),serverinfo::server_compatible );
+            servers = servers_new;
+        }
+
+        intret(servers.length());
+    }
+    else if(servers.inrange(server))
+    {
+        servers[server]->cube_get_property( prop, idx );
```

Comment:
Any reason this method name is prefixed with `cube_`? TODO for myself: check corresponding header.